### DISCUSSION TITLE
How to execute subtransactions

### DISCUSSION THREAD

**Discussion by leandrogmuller** [sqlalchemy/sqlalchemy#6325] [Usage Questions]
Hi.

How can I execute subtransactions?
Example below not work.

engine = create_engine("postgresql://user:pass@localhost:5432/datadev",poolclass=NullPool)
session_factory = sessionmaker(bind=engine)
session = session_factory()
session.begin_nested()  # when uses it dons't commit.
lang03 = session.query(Lang03).filter(Lang03.pkge == 'kernel').first()
lang03.text = 'aa'
session.merge(lang03)
session.commit()
session.close()

**Comment by CaselIT** [comment]
Hi,

Can you please provide a complete example? What does not work?

**Reply by zzzeek** [reply]
I would imagine you need to add a second commit, the commit() that's in the example will release the SAVEPOINT, then you need another one to commit the actual transaction.
